Model/Payment/AbstractMethod.php

```diff
@@ -341,25 +374,61 @@ protected function handleBankSplitAdditionalInformation(InfoInterface $payment,
             $payment->setAdditionalInformation('print_url', $bill['charges'][0]['print_url']);
             $payment->setAdditionalInformation('due_at', $bill['charges'][0]['due_at']);
         }
+
+        if ($body['payment_method_code'] === PaymentMethod::PIX) {
+            $payment->setAdditionalInformation('qrcode_original_path', $bill['charges'][0]['last_transaction']['gateway_response_fields']['qrcode_original_path']);
+            $payment->setAdditionalInformation('qrcode_path', $bill['charges'][0]['last_transaction']['gateway_response_fields']['qrcode_path']);
+            $payment->setAdditionalInformation('due_at', $bill['charges'][0]['due_at']);
+        }
     }
 
     /**
      * @param array $body
      * @param $bill
+     *
      * @return bool
      */
     private function successfullyPaid(array $body, $bill)
     {
-        if (
-            $body['payment_method_code'] === PaymentMethod::BANK_SLIP
-            || $body['payment_method_code'] === PaymentMethod::DEBIT_CARD
-            || $bill['status'] === Bill::PAID_STATUS
-            || $bill['status'] === Bill::REVIEW_STATUS
-            || reset($bill['charges'])['status'] === Bill::FRAUD_REVIEW_STATUS
-        ) {
+        if ($this->isValidPaymentMethodCode($body['payment_method_code']) || $this->isValidStatus($bill)) {
             return true;
         }
 
         return false;
     }
+
+    /**
+     * @param $paymentMethodCode
+     *
+     * @return bool
+     */
+    protected function isValidPaymentMethodCode($paymentMethodCode)
+    {
+        $paymentMethodsCode = [
+            PaymentMethod::BANK_SLIP,
+            PaymentMethod::DEBIT_CARD,
+            PaymentMethod::PIX
```

Comment:
Com a sugestão anterior, você pode remover essa verificação se o método de pagamento é PIX 😉 
```suggestion
            PaymentMethod::DEBIT_CARD
```